src/v/cluster/members_table.cc

```diff
@@ -159,6 +159,17 @@ members_table::apply(model::offset version, maintenance_mode_cmd cmd) {
         return errc::success;
     }
 
+    if (_brokers.size() < 2) {
```

Comment:
Unrelated to this change per se, but do we also need a similar check when applying a decommission command? Maybe an even heavier-handed one, eg don't decommission to below the default replication factor, or don't decommission if there are single-replica partitions hosted on the affected node?

Reply:
@andrwng yes, see the comment above "there is potentially an issue with decoms...".  Basically shrinking below 3 is already broken, and needs handling as a separate job.